### PR TITLE
[skip ci] Add more Linux workflows

### PR DESCRIPTION
**Pull request by driazati**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #60890
* **#60889**

This enables the remaining non-android Linux workflows, see #60890 for testing

Differential Revision: [D29471138](https://our.internmc.facebook.com/intern/diff/D29471138)

### CODE REVIEW

**Comment by driazati** [comment]
@driazati has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29471138).

File: .github/scripts/generate_ci_workflows.py [hunk 1]
```diff
@@ -107,36 +107,36 @@ def generate_workflow_file(
...
+    PyTorchLinuxWorkflow(
+        build_environment="pytorch-linux-xenial-py3.6-gcc7",
+        docker_image_base=f"{DOCKER_REGISTRY}/pytorch/pytorch-linux-xenial-py3.6-gcc7",
+        test_runner_type=LINUX_CPU_TEST_RUNNER,
+    ),
+    PyTorchLinuxWorkflow(
+        build_environment="pytorch-linux-xenial-py3.6-clang5-asan",
+        docker_image_base=f"{DOCKER_REGISTRY}/pytorch/pytorch-linux-xenial-py3-clang5-asan",
+        test_runner_type=LINUX_CPU_TEST_RUNNER,
+    ),
```
Comment (samestep) [inline .github/scripts/generate_ci_workflows.py:134]: shouldn't this have two shards?

File: .github/scripts/generate_ci_workflows.py [hunk 2]
```diff
@@ -147,36 +147,36 @@ def generate_workflow_file(
...
+    PyTorchLinuxWorkflow(
+        build_environment="pytorch-vulkan-linux-bionic-py3.6-clang9",
+        docker_image_base=f"{DOCKER_REGISTRY}/pytorch/pytorch-linux-bionic-py3.6-clang9",
+        test_runner_type=LINUX_CPU_TEST_RUNNER,
+    ),
+    PyTorchLinuxWorkflow(
+        build_environment="pytorch-linux-bionic-py3.8-gcc9-coverage",
+        docker_image_base=f"{DOCKER_REGISTRY}/pytorch/pytorch-linux-bionic-py3.8-gcc9",
+        test_runner_type=LINUX_CPU_TEST_RUNNER,
+    ),
```
Comment (samestep) [inline .github/scripts/generate_ci_workflows.py:174]: this one has already been merged in #61050, right?

File: .github/scripts/generate_ci_workflows.py [hunk 3]
```diff
@@ -147,36 +147,36 @@ def generate_workflow_file(
...
+    PyTorchLinuxWorkflow(
+        build_environment="pytorch-linux-bionic-rocm3.9-py3.6",
+        docker_image_base=f"{DOCKER_REGISTRY}/pytorch/pytorch-linux-bionic-rocm3.9-py3.6",
+        test_runner_type=LINUX_CPU_TEST_RUNNER,
+    ),
```
Comment (samestep) [inline .github/scripts/generate_ci_workflows.py:179]: this one isn't supposed to have a test job associated with it, is it?

File: .github/scripts/generate_ci_workflows.py [hunk 4]
```diff
@@ -107,36 +107,36 @@ def generate_workflow_file(
...
+    PyTorchLinuxWorkflow(
+        build_environment="pytorch-linux-xenial-py3.6-clang5-asan",
+        docker_image_base=f"{DOCKER_REGISTRY}/pytorch/pytorch-linux-xenial-py3-clang5-asan",
+        test_runner_type=LINUX_CPU_TEST_RUNNER,
+    ),
+    PyTorchLinuxWorkflow(
+        build_environment="pytorch-linux-xenial-py3.6-clang7-onnx",
+        docker_image_base=f"{DOCKER_REGISTRY}/pytorch/pytorch-linux-xenial-py3-clang7-onnx",
+        test_runner_type=LINUX_CPU_TEST_RUNNER,
+    ),
```
Comment (samestep) [inline .github/scripts/generate_ci_workflows.py:139]: I think this one should also have two shards if I'm not mistaken

File: .github/scripts/generate_ci_workflows.py [hunk 5]
```diff
@@ -107,36 +107,36 @@ def generate_workflow_file(
...
+    PyTorchLinuxWorkflow(
+        build_environment="pytorch-parallelnative-linux-xenial-py3.6-gcc5.4",
+        docker_image_base=f"{DOCKER_REGISTRY}/pytorch/pytorch-linux-xenial-py3.6-gcc5.4",
+        test_runner_type=LINUX_CPU_TEST_RUNNER,
+    ),
+    PyTorchLinuxWorkflow(
+        build_environment="pytorch-pure_torch-linux-xenial-py3.6-gcc5.4",
+        docker_image_base=f"{DOCKER_REGISTRY}/pytorch/pytorch-linux-xenial-py3.6-gcc5.4",
+        test_runner_type=LINUX_CPU_TEST_RUNNER,
+    ),
```
Comment (samestep) [inline .github/scripts/generate_ci_workflows.py:124]: I think this one also isn't supposed to have a test job attached to it

File: .github/scripts/generate_ci_workflows.py [hunk 6]
```diff
@@ -147,36 +147,36 @@ def generate_workflow_file(
...
-    # PyTorchLinuxWorkflow(
-    #     build_environment="pytorch-linux-bionic-rocm3.9-py3.6",
-    #     docker_image_base=f"{DOCKER_REGISTRY}/pytorch/pytorch-linux-bionic-rocm3.9-py3.6",
-    #     test_runner_type=LINUX_CPU_TEST_RUNNER,
-    # ),
+    PyTorchLinuxWorkflow(
+        build_environment="pytorch-libtorch-linux-xenial-cuda11.1-cudnn8-py3.6-gcc7",
+        docker_image_base=f"{DOCKER_REGISTRY}/pytorch/pytorch-linux-xenial-cuda11.1-cudnn8-py3-gcc7",
+        test_runner_type=LINUX_CUDA_TEST_RUNNER,
+    ),
```
Comment (samestep) [inline .github/scripts/generate_ci_workflows.py:154]: same for this one regarding not having a test job

**Comment by samestep** [comment]
> This enables the remaining non-android Linux workflows

According to #57686, I think you're missing a few:

- pytorch-linux-xenial-py3.6-gcc5.4
  - backward-compatibility-check
  - cpp-doc
  - python-doc-push
  - doc-test
  - jit-legacy-test
- pytorch-libtorch-linux-xenial-cuda10.2-cudnn7-py3.6-gcc7

**Pull request closed without merging**